db/models/models.js

```diff
@@ -1,26 +1,49 @@
 const db = require("../connection.js");
+const categories = require("../data/test-data/categories.js");
 const { checkExists } = require("../utils.js");
 
 exports.fetchCategories = () => {
   return db.query(`SELECT * FROM categories;`).then((results) => results.rows);
 };
-exports.fetchReviews = () => {
-  return db
-    .query(
-      `SELECT users.username AS "owner", title, reviews.review_id,
+
+exports.fetchReviews = (category, sort_by = "created_at", order = "DESC") => {
+  const validColumns = [
+    "owner",
+    "title",
+    "review_id",
+    "category",
+    "review_img_url",
+    "created_at",
+    "votes",
+    "designer",
+    "comment_count",
+  ];
+  if (!validColumns.includes(sort_by)) {
+    return Promise.reject({ status: 400, msg: "invalid sort query" });
+  }
+  let queryStr = `SELECT reviews.owner, reviews.title, reviews.review_id,
       reviews.category, reviews.review_img_url,
       reviews.created_at, reviews.votes, reviews.designer,
       CAST(COALESCE(COUNT(comments.review_id), 0) AS INT) AS "comment_count"
       FROM reviews
-      JOIN users ON users.username = reviews.owner
       FULL OUTER JOIN comments ON comments.review_id = reviews.review_id
-      GROUP BY users.username, reviews.review_id
-      ORDER BY reviews.created_at DESC;
-        `
-    )
-    .then((results) => {
-      return results.rows;
-    });
+      `;
+  if (category != undefined) {
+    queryStr += `WHERE reviews.category = '${category}'`;
```

Comment:
You're open to sql injection here! MUST use parameterised query $ syntax instead